### DISCUSSION TITLE
Does this project need a new maintainer?

### DISCUSSION THREAD

**Discussion by qtzar** [thomvaill/log4brains#108] [General]
There hasn't been any updates in over 6 months and there are a lot of outstanding PRs and issues.

**Comment by ejwill** [comment]
@qtzar how would we go about doing that?

**Comment by Potherca** [comment]
The first step would be to reach out to the original maintainer and ask them to add other people as co-maintainer.

If GitHub is not responded to, whichever other channels will need to be found and used.

The issue is usually finding other people who have both the knowledge, motivation and time required to (help) maintain an open-source product.

Related: https://github.com/thomvaill/log4brains/issues/29#issuecomment-829962251 by @thomvaill

**Reply by evanerwee** [reply]
Let's start the process. I'm very interested in helping where I can. My team did look at the code purely from a maintenance perspective and they agree that it will take about 3 weeks to update dependencies and get to a 'more' stable Node18 release, but a while from 'perfect'. The first step is to find new maintainers.

**Reply by Potherca** [reply]
There is a contact form on https://bluebricks.dev/ that might yield results and we could try https://twitter.com/thomvaill (although that seems mostly silent?) or https://www.linkedin.com/in/tvaillant/

If you feel serious enough about investing future energy and time on this, the next logical step would be to create a community organization for `log4brains`. However, it looks like that username is already taken on GitHub so another name might be needed.

We could start/open a GH discussion there (organisation level) to discuss what needs doing, keep ourselves and others informed of progress, compose a message for @thomvaill, etc.

**Reply by evanerwee01** [reply]
The project represents a simple and straightforward way to create ADRs. It is not the end-of-the-world to create another avenue. What I do not know is the consensus. Was the project helpful, was it used? I will see if I can get some questions answered.

**Reply by evanerwee01** [reply]
Sorry, I have a dozen GitID's thanks to my organization. Just authenticated using my work ID. Stick to evanerwee, that is my private ID.

**Reply by philCryoport** [reply]
@evanerwee  / @evanerwee01 / @Potherca was a community fork ever created?  If so, where can I find it?

**Comment by thomvaill** [comment]
Hello !
Thank you so much for your interest in the project and I apologize a lot for being silent the last months.
To be honest with you my wife and I had our first child last September, and that's one of the reasons I did not manage to dedicate some time to the project.
I've already been contacted by email by a contributor who volunteered to help maintaining the project.
I really appreciate your contributions and obviously some help would be very welcome to help me maintain the project.
I will try to dedicate some time by the end of the year to review the pending PRs and organize the project in a way I am not a bottleneck anymore.
By the way, if you have some recommendations or documentation on this topic, you're very welcome.

**Reply by philCryoport** [reply]
Hi @thomvaill congrats on your first child!  When my wife and I had twins, our lives basically stopped, so fully understand.

Did that contributor you mention above ever accept the role of maintaining the project?  There's 11 open PRs as of this writing...

**Reply by thomvaill** [reply]
Hi @philCryoport thank you for your support :-)
I didn't have the opportunity to talk again with this contributor since our last conversation.
I am currently working on adapting the CI/CD of the project to be able to release RC candidates automatically, so it will  allow us to merge the merge requests more quickly, so that our power users can test the RC candidate.
After that, I will open the discussion again with potential mainteners

**Reply by jean-francoismougnot** [reply]
Hi @thomvaill,
Sorry to bother and bump this topic again and thank you for the awesome initiative and your child 👍 
I believe we do reach a critical situation with almost 2 years with outdated deps, no nodejs bumps, and end of support of [node16 in github actions announced](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/), itself being out of support since last September.

I believe contributors have done what's need to be done and we're almost here, we just need a merge and release at least in order to avoid huge CVE for anyone using this awesome tool.

I, with my team, can offer help in order to contribute and maintain to this project since we rely heavily on OSS and believe it's our responsibility to make such project live as long as possible by helping.

Other option is to fork, but we rather prefer put our energy in this collective and community approach than doing pretty much the same thing needed here, but on "our side" meaning double lifecycle to maintain basicaly.

Can you please let me know what do you think / plan (i'm also a french guy for what it's worth 🇫🇷 🤓 ) ?

**Comment by evanerwee01** [comment]
It is truly an amazing product. I’m more than happy to perform testing. I’m the Architecture Capability Lead for Tech Assets at Deloitte USA.

From: jean-francoismougnot ***@***.***>
Date: Wednesday, September 4, 2024 at 8:50 AM
To: thomvaill/log4brains ***@***.***>
Cc: Erwee, Evan ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [thomvaill/log4brains] Does this project need a new maintainer? (Discussion #108)

Hi @thomvaill<https://github.com/thomvaill>,
Sorry to bother and bump this topic again and thank you for the awesome initiative and your child 👍
I believe we do reach a critical situation with almost 2 years with outdated deps, no nodejs bumps, and end of support of node16 in github actions announced<https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/>, itself being out of support since last September.

I believe contributors have done what's need to be done and we're almost here, we just need a merge and release at least in order to avoid huge CVE for anyone using this awesome tool.

I, with my team, can offer help in order to contribute and maintain to this project since we rely heavily on OSS and believe it's our responsibility to make such project live as long as possible by helping.

Other option is to fork, but we rather prefer put our energy in this collective and community approach than doing pretty much the same thing needed here, but on "our side" meaning double lifecycle to maintain basicaly.

Can you please let me know what do you think / plan (i'm also a french guy for what it's worth 🇫🇷 🤓 ) ?

—
Reply to this email directly, view it on GitHub<https://github.com/thomvaill/log4brains/discussions/108#discussioncomment-10545389>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUPKU3BR5UMXTZ7BN4RRXVTZU36ZZAVCNFSM6AAAAAAYD65VO6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJUGUZTQOI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

This message (including any attachments) contains confidential information intended for a specific individual and purpose, and is protected by law. If you are not the intended recipient, you should delete this message and any disclosure, copying, or distribution of this message, or the taking of any action based on it, by you is strictly prohibited.

Deloitte refers to a Deloitte member firm, one of its related entities, or Deloitte Touche Tohmatsu Limited ("DTTL"). Each Deloitte member firm is a separate legal entity and a member of DTTL. DTTL does not provide services to clients. Please see www.deloitte.com/about to learn more.

v.E.1

**Comment by thomvaill** [comment]
Hi everyone! First, I owe you all an apology for my year-long silence.
Life got in the way (but for good reasons 👶!!), but I’m excited to dive back into Log4brains and improve it for the community!

As discussed earlier, I was contacted by some of you that were interested in contributing, and becoming co-maintainer of the project.
I really appreciate your support and I took some time to think about how to facilitate these contributions, and to be able to share my responsibilities.
Also, as you highlighted: there are many pending merge requests waiting to be reviewed and merged.

Therefore, here is my proposal of our new way of working together: https://github.com/thomvaill/log4brains/pull/121
I hope it includes both the solution to streamline contributions and to define a way for motivated contributors to become maintainers (see the ["Would like to become a co-maintainer?" paragraph of the CONTRIBUTING.md](https://github.com/thomvaill/log4brains/pull/121/files?short_path=eca12c0#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055) in the pull request)
I’d love to hear your feedback on it! This is still a WIP.


**Next steps on my side:** Over the coming weeks, I will do my best to rebase and merge the current open pull requests, prioritizing security updates. I greatly appreciate your patience and contributions!

**Next steps for those who volunteered to help:** Review the new workflow proposal, give me your feedback, and see the "Would like to become a co-maintainer?" paragraph of the CONTRIBUTING.md. Thanks a lot!! 🙏 

Looking forward to hearing your thoughts and getting back to building Log4brains together!

**Reply by Potherca** [reply]
Awesome! And congratulation! And good to have you back! Hope you are doing well. :hugs:

**Comment by evanerwee01** [comment]
First, thank you. Second,  I will review the links proposals more closely tomorrow.
________________________________
From: Thomas Vaillant ***@***.***>
Sent: Thursday, September 26, 2024 5:42:17 PM
To: thomvaill/log4brains ***@***.***>
Cc: Erwee, Evan ***@***.***>; Mention ***@***.***>
Subject: [EXT] Re: [thomvaill/log4brains] Does this project need a new maintainer? (Discussion #108)


Hi everyone! First, I owe you all an apology for my year-long silence.
Life got in the way (but for good reasons 👶!!), but I’m excited to dive back into Log4brains and improve it for the community!

As discussed earlier, I was contacted by some of you that were interested in contributing, and becoming co-maintainer of the project.
I really appreciate your support and I took some time to think about how to facilitate these contributions, and to be able to share my responsibilities.
Also, as you highlighted: there are many pending merge requests waiting to be reviewed and merged.

Therefore, here is my proposal of our new way of working together: #121<https://github.com/thomvaill/log4brains/pull/121>
I hope it includes both the solution to streamline contributions and to define a way for motivated contributors to become maintainers (see the "Would like to become a co-maintainer?" paragraph of the CONTRIBUTING.md<https://github.com/thomvaill/log4brains/pull/121/files?short_path=eca12c0#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055> in the pull request)
I’d love to hear your feedback on it! This is still a WIP.

Next steps on my side: Over the coming weeks, I will do my best to rebase and merge the current open pull requests, prioritizing security updates. I greatly appreciate your patience and contributions!

Next steps for those who volunteered to help: Review the new workflow proposal, give me your feedback, and see the "Would like to become a co-maintainer?" paragraph of the CONTRIBUTING.md. Thanks a lot!! 🙏

Looking forward to hearing your thoughts and getting back to building Log4brains together!

—
Reply to this email directly, view it on GitHub<https://github.com/thomvaill/log4brains/discussions/108#discussioncomment-10768555>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUPKU3FEB3SNJ5L5BOJ7AADZYR5TTAVCNFSM6AAAAAAYD65VO6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZWHA2TKNI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

This message (including any attachments) contains confidential information intended for a specific individual and purpose, and is protected by law. If you are not the intended recipient, you should delete this message and any disclosure, copying, or distribution of this message, or the taking of any action based on it, by you is strictly prohibited.

Deloitte refers to a Deloitte member firm, one of its related entities, or Deloitte Touche Tohmatsu Limited ("DTTL"). Each Deloitte member firm is a separate legal entity and a member of DTTL. DTTL does not provide services to clients. Please see www.deloitte.com/about to learn more.

v.E.1

**Comment by thomvaill** [comment]
Hi all!

I am very happy to announce you that [v1.1.0](https://github.com/thomvaill/log4brains/releases/tag/v1.1.0) is finally released after a very long pause :-)
There is still much work to do but at least the app is now running on the latest Node.js version.

And thanks to the new [CONTRIBUTING.md](https://github.com/thomvaill/log4brains/blob/develop/CONTRIBUTING.md) process, I am sure we will be able to release more often :-)

I think I can close this discussion ;)

**Comment by evanerwee01** [comment]
Thank You
________________________________
From: Thomas Vaillant ***@***.***>
Sent: Tuesday, December 17, 2024 1:30:05 PM
To: thomvaill/log4brains ***@***.***>
Cc: Erwee, Evan ***@***.***>; Mention ***@***.***>
Subject: [EXT] Re: [thomvaill/log4brains] Does this project need a new maintainer? (Discussion #108)


Hi all!

I am very happy to announce you that v1.1.0<https://github.com/thomvaill/log4brains/releases/tag/v1.1.0> is finally released after a very long pause :-)
There is still much work to do but at least the app is now running on the latest Node.js version.

And thanks to the new CONTRIBUTING.md<https://github.com/thomvaill/log4brains/blob/develop/CONTRIBUTING.md> process, I am sure we will be able to release more often :-)

I think I can close this discussion ;)

—
Reply to this email directly, view it on GitHub<https://github.com/thomvaill/log4brains/discussions/108#discussioncomment-11597003>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUPKU3EORQ5WYW7QRCSY6W32GBUS3AVCNFSM6AAAAAAYD65VO6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJZG4YDAMY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

This message (including any attachments) contains confidential information intended for a specific individual and purpose, and is protected by law. If you are not the intended recipient, you should delete this message and any disclosure, copying, or distribution of this message, or the taking of any action based on it, by you is strictly prohibited.

Deloitte refers to a Deloitte member firm, one of its related entities, or Deloitte Touche Tohmatsu Limited ("DTTL"). Each Deloitte member firm is a separate legal entity and a member of DTTL. DTTL does not provide services to clients. Please see www.deloitte.com/about to learn more.

v.E.1

**Comment by saksham-netlight** [comment]
Thank you!
